sponsorship_compassion/models/contracts.py

```diff
@@ -249,7 +249,37 @@ def create(self, vals):
                 'sponsor_id': vals.get('correspondent_id', vals['partner_id'])
             })
 
-        return super(SponsorshipContract, self).create(vals)
+        """ Generates commitment number for contracts BVRs
+        """
+        if 'commitment_number' not in vals:
+            partner_id = vals.get('partner_id')
+            correspondent_id = vals.get('correspondent_id')
+            if partner_id:
+                other_nums = self.search([
+                    '|', '|', '|', ('partner_id', '=', partner_id),
+                    ('partner_id', '=', correspondent_id),
+                    ('correspondent_id', '=', partner_id),
+                    ('correspondent_id', '=', correspondent_id)
+                ]).mapped('commitment_number')
+
+                vals['commitment_number'] = max(other_nums or [-1]) + 1
+            else:
+                vals['commitment_number'] = 1
+
+        new_sponsorship = super(SponsorshipContract, self).create(vals)
```

Comment:
You can directly return super at this point and remove the last part which will stay in contract_compassion module.